go/cmd/vtgate/cli/cli.go

```diff
@@ -146,7 +146,9 @@ func run(cmd *cobra.Command, args []string) error {
 	ts := topo.Open()
 	defer ts.Close()
 
-	resilientServer = srvtopo.NewResilientServer(context.Background(), ts, srvTopoCounts)
+	ctx, cancel := context.WithCancel(cmd.Context())
+	defer cancel()
+	resilientServer = srvtopo.NewResilientServer(ctx, ts, srvTopoCounts)
```

Comment:
This fix ensures that during shutdown we don't hit the panic from #15927, because the context will have been cancelled always before the topo is closed (which is deferred right above here).